### DISCUSSION TITLE
How to use a custom route parameter based Eloquent/Table Query with filters

### DISCUSSION THREAD

**Discussion by supungs** [filamentphp/filament#6879] [General]
I have Project and Expense resources where expenses belongs to a project. So, I have a custom route for the Expense resource as programs/{project}/expenses to display only the Expenses related to a particular project. I have the getEloquentQuery() like below and I get the expected result on initial page load.

```php
   public static function getEloquentQuery(): Builder
    {
        if (request('project')) {
            return parent::getEloquentQuery()->where('project_id', request('project'));
        } else {
            return parent::getEloquentQuery();
        }
    }
```

But the problem comes when I use Filters in the table. The filtered results come without considering the `->where('project_id', request('project'))`, so I get expenses from all projects. In fact when the filtered result request is made, `request('project')` returns `null` within the `getEloquentQuery()` method.

What is the proper way to do this? How can I access the route parameter in the filtered result request?

**Comment by danharrin** [comment]
On the list page, you need to override `mount()` and save the value of `project` into a public property, which will be persisted by Livewire between requests. Then, instead of overriding `getEloquentQuery()` on the resource class, override `getTableQuery()` on the list page class and access your public property there.

**Reply by supungs** [reply]
It works. Thank you so much..

```php
    public $project = null; 

    public function mount(): void
    {
        static::authorizeResourceAccess();
        $this->project = Project::find(request('project'));
        abort_unless($this->project != null, 404);
    }

    protected function getTableQuery(): Builder
    {
        return Expense::where('project_id', $this->project->id);
    }
```
